src/components/Calendar/Calendar.js

```diff
@@ -152,16 +151,25 @@ const MyCalendar = (props) => {
         student: e.student,
         resourceId: e.resourceId,
         title: e.title,
-        location: e.location
+        location: e.location,
       });
       showModal();
     }
   };
 
+  const filterEventsByLocation = (location) => {
+    if (location === '') {
+      setAllEvents(events);
+    } else {
+      const filteredEvents = allEvents.filter((e) => e.location === location);
```

Comment:
I was testing it out and I think it needs a small tweak. Before the list is filtered it needs to be reset to the setAllEvents needs to be reset to events and then filtered. Otherwise, you are filtering from an already filtered list. For example, if you filter by India and then change the filter to Africa you will get no results because it is only filtering from the filtered India list.

Reply:
Ah, thank you for catching that and for the help too. I already made the changes.